temporal-sdk/src/main/java/io/temporal/internal/sync/POJOWorkflowImplementationFactory.java

```diff
@@ -349,22 +363,30 @@ public WorkflowOutput execute(WorkflowInput input) {
         }
       }
 
-      protected void newInstance() {
+      protected void newInstance(Optional<Payloads> input) {
         Func<?> factory = workflowInstanceFactories.get(workflowImplementationClass);
         if (factory != null) {
           workflow = factory.apply();
         } else {
           try {
-            workflow = workflowImplementationClass.getDeclaredConstructor().newInstance();
+            try {
+              workflow = workflowImplementationClass.getDeclaredConstructor().newInstance();
```

Comment:
Do we validate other things at worker registration time with regards to workflow structure and annotation use? Meaning something like no duplicate signals? Can we validate the constructor there and also know which one we're using? I think we should validate that either no constructor is present, a parameterless constructor is present, or a constructor matching the workflow run parameters is present. And if _both_ parameterless and parameter-based constructors are present, we should log a warning IMO since that we choose the parameterless may not be obvious in ambiguous situations.

Arguably we should warn if there is more than one public constructor no matter what. IMO there are not very many good use cases for a non-Temporal-invoked constructor on a workflow. It'd either be a user wanting to instantiate themselves somehow in a test (but we need to control the environment for testing workflow code) or a user having a constructor that may be invoked by a subclass (they should mark it protected).

Reply:
Yeah I hesitate to add validation where it didn't exist before, but if it is just warning that is probably fine and useful. Libraries normally only require an annotation if their is some ambiguity, Jackson does not require `@JsonCreator` you can have annotationless constructors.

Reply:
Right, IMO the annotation should only be required if you are leaving our long-standing default behavior. You are basically opting in to a special type of constructor.

Reply:
I think only reason you would require an annotation is if their is ambiguity. The only case I can think of where there is ambiguity is if both a no-arg and argument constructor exist. So if we want to add an annotation and require it in the ambiguous case I'm OK with that just seems like a lot of extra work for a niche case.

Reply:
That a user has to think about whether they are ambiguous or not for the annotation makes it less worth it. I think it's simpler for a user to understand it's always required if they want the args.